src/EventStore.ClientAPI/Projections/ProjectionDetails.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+
+namespace EventStore.ClientAPI.Projections
+{
+    /// <summary>
+    /// Provides the details for a projection.
+    /// </summary>
+    public sealed class ProjectionDetails
+    {
+        /// <summary>
+        /// The CoreProcessingTime
+        /// </summary>
+        public readonly int CoreProcessingTime;
+
+        /// <summary>
+        /// The projection version
+        /// </summary>
+        public readonly int Version;
+
+        /// <summary>
+        /// The Epoch
+        /// </summary>
+        public readonly int Epoch;
+
+        /// <summary>
+        /// The projection EffectiveName
+        /// </summary>
+        public readonly string EffectiveName;
+
+        /// <summary>
+        /// The projection WritesInProgress
+        /// </summary>
+        public readonly int WritesInProgress;
+
+        /// <summary>
+        /// The projection ReadsInProgress
+        /// </summary>
+        public readonly int ReadsInProgress;
+
+        /// <summary>
+        /// The projection PartitionsCached
+        /// </summary>
+        public readonly int PartitionsCached;
+
+        /// <summary>
+        /// The projection Status
+        /// </summary>
+        public readonly string Status;
+
+        /// <summary>
+        /// The projection StateReason
+        /// </summary>
+        public readonly string StateReason;
+
+        /// <summary>
+        /// The projection Name
+        /// </summary>
+        public readonly string Name;
+
+        /// <summary>
+        /// The projection Mode
+        /// </summary>
+        public readonly string Mode;
+
+        /// <summary>
+        /// The projection Position
+        /// </summary>
+        public readonly string Position;
+
+        /// <summary>
+        /// The projection Progress
+        /// </summary>
+        public readonly decimal Progress;
+
+        /// <summary>
+        /// LastCheckpoint
+        /// </summary>
+        public readonly string LastCheckpoint;
```

Comment:
We could possibly setup another structure that shows Stream/Position and provide an array
Something along the lines of
```
StreamPosition[] streamPositions
```
Also, could be part of another PR.